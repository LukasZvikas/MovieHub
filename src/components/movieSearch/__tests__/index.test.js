import { shallowMount, mount } from "@vue/test-utils";

import MovieSearchWrapper from "../";

jest.mock("../../../utilities/fetch");

const search_data = [
  {
    adult: false,
    backdrop_path: "/wqtaHWOEZ3rXDJ8c6ZZShulbo18.jpg",
    id: 332562,
    overview:
      "Seasoned musician Jackson Maine discovers—and falls in love with—struggling artist Ally. She has just about given up on her dream to make it big as a singer—until Jack coaxes her into the spotlight. But even as Ally's career takes off, the personal side of their relationship is breaking down, as Jack fights an ongoing battle with his own internal demons.",
    poster_path: "/wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg",
    release_date: "2018-10-03",
    title: "A Star Is Born",
    vote_average: 7.5
  }
];

describe("Movie Search if input query has a result and fetch are triggered", () => {
  let wrapper;
  beforeEach(done => {
    wrapper = mount(MovieSearchWrapper);
    const input = wrapper.find("input");
    input.element.value = "star";
    input.trigger("input");
    wrapper.vm.$nextTick(() => {
      wrapper.setData({ searchResult: [...search_data] });
      done();
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });
  it("Search results has correct heading", () => {
    expect(
      wrapper.find('[data-test="movie-search-query-result-heading"]').text()
    ).toBe('Movies matching: "star"');
  });
  it("movie box item img has a correct src", () => {
    expect(
      wrapper
        .find('[data-test="movie-item-search-box-image"]')
        .attributes("src")
    ).toBe(
      "http://image.tmdb.org/t/p/original//wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
    );
  });

  it("movie box item has a correct movie title", () => {
    expect(
      wrapper.find('[data-test="movie-item-search-box-title"]').text()
    ).toBe("A Star Is Born");
  });

  it("there is one result item", () => {
    expect(
      wrapper.findAll('[data-test="movie-item-search-result-item"]')
    ).toHaveLength(1);
  });
});

describe("Movie Search if input query has not result and fetch are triggered", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(MovieSearchWrapper);
    const input = wrapper.find("input");

    input.element.value = "starisborn";
    input.trigger("input");
    wrapper.vm.$nextTick(() => {
      wrapper.setData({ searchResult: [] });
      expect(wrapper.vm.$el).toMatchSnapshot();
    });
  });
  it("Search results has correct heading", () => {
    expect(wrapper.find('[data-test="movie-search-heading"]').text()).toBe(
      "Ooops!"
    );
  });
  it("Search results has correct body", () => {
    expect(wrapper.find('[data-test="movie-search-body"]').text()).toBe(
      "It appears that the movie your are looking for is not available."
    );
  });
  it("there is one result item", () => {
    expect(
      wrapper.findAll('[data-test="movie-item-search-result-item"]')
    ).toHaveLength(0);
  });
});
