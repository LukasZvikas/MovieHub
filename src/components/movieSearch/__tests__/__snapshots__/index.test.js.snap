// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Search if input query has a result and fetch are triggered Search results has correct heading 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="movie-list"
      data-test="movie-search-background-image"
    >
      <div>
        <div
          class="search-term"
          data-test="movie-search-query-result-heading"
        >
          Movies matching: "star"
        </div>
         
        <div
          class="row"
        >
          <div
            class="movie-item-box"
            data-test="movie-item-search-result-item"
          >
            <img
              class="img-fluid img-cover"
              data-test="movie-item-search-box-image"
              src="http://image.tmdb.org/t/p/original//wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
            />
             
            <div
              class="movie-item-box-title"
              data-test="movie-item-search-box-title"
            >
              A Star Is Born
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie Search if input query has a result and fetch are triggered movie box item has a correct movie title 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="movie-list"
      data-test="movie-search-background-image"
    >
      <div>
        <div
          class="search-term"
          data-test="movie-search-query-result-heading"
        >
          Movies matching: "star"
        </div>
         
        <div
          class="row"
        >
          <div
            class="movie-item-box"
            data-test="movie-item-search-result-item"
          >
            <img
              class="img-fluid img-cover"
              data-test="movie-item-search-box-image"
              src="http://image.tmdb.org/t/p/original//wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
            />
             
            <div
              class="movie-item-box-title"
              data-test="movie-item-search-box-title"
            >
              A Star Is Born
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie Search if input query has a result and fetch are triggered movie box item img has a correct src 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="movie-list"
      data-test="movie-search-background-image"
    >
      <div>
        <div
          class="search-term"
          data-test="movie-search-query-result-heading"
        >
          Movies matching: "star"
        </div>
         
        <div
          class="row"
        >
          <div
            class="movie-item-box"
            data-test="movie-item-search-result-item"
          >
            <img
              class="img-fluid img-cover"
              data-test="movie-item-search-box-image"
              src="http://image.tmdb.org/t/p/original//wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
            />
             
            <div
              class="movie-item-box-title"
              data-test="movie-item-search-box-title"
            >
              A Star Is Born
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie Search if input query has a result and fetch are triggered there is one result item 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="movie-list"
      data-test="movie-search-background-image"
    >
      <div>
        <div
          class="search-term"
          data-test="movie-search-query-result-heading"
        >
          Movies matching: "star"
        </div>
         
        <div
          class="row"
        >
          <div
            class="movie-item-box"
            data-test="movie-item-search-result-item"
          >
            <img
              class="img-fluid img-cover"
              data-test="movie-item-search-box-image"
              src="http://image.tmdb.org/t/p/original//wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
            />
             
            <div
              class="movie-item-box-title"
              data-test="movie-item-search-box-title"
            >
              A Star Is Born
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie Search if input query has not result and fetch are triggered Search results has correct body 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="image back-img-not-found"
      data-test="movie-search-background-image"
    >
      <div
        class="search-text-wrap"
      >
        <div>
          <div
            class="search-text-heading"
            data-test="movie-search-heading"
          >
            Ooops!
          </div>
           
          <div
            class="search-text-body"
            data-test="movie-search-body"
          >
            It appears that the movie your are looking for is not available.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie Search if input query has not result and fetch are triggered Search results has correct heading 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="image back-img-not-found"
      data-test="movie-search-background-image"
    >
      <div
        class="search-text-wrap"
      >
        <div>
          <div
            class="search-text-heading"
            data-test="movie-search-heading"
          >
            Ooops!
          </div>
           
          <div
            class="search-text-body"
            data-test="movie-search-body"
          >
            It appears that the movie your are looking for is not available.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie Search if input query has not result and fetch are triggered there is one result item 1`] = `
<div
  class="row"
  style="height: 90vh;"
>
  <div
    class="col-sm-12 col-md-3"
  >
    <div
      class="input-group mb-3 search-bar"
    >
      <div
        class="search-bar items-wrap"
      >
        <input
          class="form-control"
          placeholder="Search movies"
        />
         
        <div
          class="search-bar circle d-flex justify-content-center align-items-center"
        >
          <svg
            fill="whitesmoke"
            height="25"
            version="1.1"
            viewBox="0 0 20 20"
            width="25"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.545 15.467l-3.779-3.779c0.57-0.935 0.898-2.035 0.898-3.21 0-3.417-2.961-6.377-6.378-6.377s-6.186 2.769-6.186 6.186c0 3.416 2.961 6.377 6.377 6.377 1.137 0 2.2-0.309 3.115-0.844l3.799 3.801c0.372 0.371 0.975 0.371 1.346 0l0.943-0.943c0.371-0.371 0.236-0.84-0.135-1.211zM4.004 8.287c0-2.366 1.917-4.283 4.282-4.283s4.474 2.107 4.474 4.474c0 2.365-1.918 4.283-4.283 4.283s-4.473-2.109-4.473-4.474z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="col-sm-12 col-md-9"
    style="overflow: auto;"
  >
    <div
      class="image back-img-not-found"
      data-test="movie-search-background-image"
    >
      <div
        class="search-text-wrap"
      >
        <div>
          <div
            class="search-text-heading"
            data-test="movie-search-heading"
          >
            Ooops!
          </div>
           
          <div
            class="search-text-body"
            data-test="movie-search-body"
          >
            It appears that the movie your are looking for is not available.
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
